docs/articles/persistence/custom-persistence-provider.md

```diff
@@ -0,0 +1,467 @@
+---
+uid: custom-persistent-provider
+title: Writing A Custom Persistent Provider
+---
+
+# Writing A Custom Akka.Persistence Provider
+
+For an introduction to event sourcing, you can read this [article](https://docs.microsoft.com/en-us/previous-versions/msp-n-p/jj591559%28v=pandp.10%29) at MSDN.
+
+Persistence or event source provider in Akka.NET are divided into two plugins, the journal and snapshot store plugins, each with their own API and responsibility. The goal of `Akka.Persistence` event sourcing plugin is to provide a consistent event sourcing API that provides an abstraction layer over the underlying storage mechanism.
+
+## Implementing Akka.Persistence Journal
+
+Akka.Persistence journal is responsible for storing all events for playback in the event of a recovery.
+
+All of the code examples in this documentation will assume a SQLite database. You can view the complete sample project in `/src/examples/Akka.Persistence.Custom`.
+
+The `Journal` table schemas we will be using are:
+
+[!code-csharp[Schema](../../../src/examples/Akka.Persistence.Custom/Journal/SqliteJournal.cs?name=schema "Journal table schemas")]
+
+### Required Method Implementations
+
+#### ReplayMessagesAsync
+
+```c#
```

Comment:
Do we leave this as it is or make use of the range selection though the line number may change in the future?

Reply:
That is the name of the function that needs to be implemented, the code block is used here just for formatting